### PR TITLE
add fortran compiler

### PR DESCRIPTION
**Pull request by utsekaj42**
Checklist
* [ X] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Wasn't sure how to request a new branch (3.1.1).

I assume adding the fortran compiler wouldn't hurt the other branches, but I'm intending this to improve distribution of a package that depends on sundials 3.1.1 with fortran support.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% set name = "sundials" %}
-{% set version = "4.1.0" %}
+{% set version = "3.1.1" %}
```
Comment (jschueller) [inline recipe/meta.yaml]: you're going back to 3.1

**Comment by jschueller** [comment]
- do you explicitly need 3.1 ? else there's a 3.2 branch
- seems fortran support needs to be enabled via a cmake option (see build.sh/bld.bat)

**Comment by utsekaj42** [comment]
Hmm, I need to do some more testing locally. I thought the build I had worked, but on further 
inspection it seems not have.

I am not entirely sure if there are breaking changes from 3.1.1 to 3.2 but the package, scikits.odes, only supports 3.1.1 .

Outside of the feed-stock, building with
```
cmake \
-DLAPACK_ENABLE=ON \
 -DSUNDIALS_INDEX_TYPE=int32_t \
-DCMAKE_INSTALL_PREFIX=$CONDA_PREFIX ../sundials-3.1.1/  
make install
```
is sufficient to then install scikits.odes, but installing the package resulting from 'build-locally.py' doesn't seem to work the same.

**Comment by bjodah** [comment]
@utsekaj42 have you tried with 3.2.1?
Looks like nix uses it:
https://github.com/flokli/nixpkgs/commit/7e9e22f4d57816a74a7050a8edc0e85d63427fea

**Comment by jschueller** [comment]
outdated

**Pull request closed without merging**